rpc/src/test/scala/org/bitcoins/rpc/RpcClientTest.scala

```diff
@@ -0,0 +1,517 @@
+package org.bitcoins.rpc
+
+import akka.actor.ActorSystem
+import akka.stream.ActorMaterializer
+import org.bitcoins.core.config.RegTest
+import org.bitcoins.core.crypto.DoubleSha256Digest
+import org.bitcoins.core.currency
+import org.bitcoins.core.currency.{Bitcoins, Satoshis}
+import org.bitcoins.core.protocol.transaction.{
+  Transaction,
+  TransactionInput,
+  TransactionOutPoint
+}
+import org.bitcoins.core.util.BitcoinSLogger
+import org.bitcoins.rpc.client.RpcClient
+import org.scalatest.AsyncFlatSpec
+import org.bitcoins.core.number.UInt32
+import org.bitcoins.core.protocol.BitcoinAddress
+import org.bitcoins.core.protocol.script.ScriptSignature
+import org.bitcoins.rpc.jsonmodels.GetTransactionResult
+
+import scala.concurrent.Future
+
+class RpcClientTest extends AsyncFlatSpec {
+  implicit val system = ActorSystem()
+  implicit val m = ActorMaterializer()
+  implicit val ec = m.executionContext
+  implicit val regTestNetworkParam = RegTest
+
+  val client = new RpcClient
+  val logger = BitcoinSLogger.logger
+
+  behavior of "RpcClient"
+
+  it should "be able to get blockchain info" in {
+    client.getBlockChainInfo.flatMap { info =>
+      logger.info(info.toString)
+      assert(info.chain == "regtest")
+      client.getBestBlockHash.map(bestHash =>
+        assert(info.bestblockhash == bestHash))
+      assert(info.softforks.length >= 3)
+      assert(info.bip9_softforks.keySet.size >= 2)
+    }
+  }
+
+  it should "be able to create a raw transaction" in {
+    client.generate(2).flatMap { blocks =>
+      client.getBlock(blocks(0)).flatMap { block0 =>
+        client.getBlock(blocks(1)).flatMap { block1 =>
+          client.getTransaction(block0.tx(0)).flatMap { transaction0 =>
+            client.getTransaction(block1.tx(0)).flatMap { transaction1 =>
+              val input0 =
+                TransactionOutPoint(transaction0.txid,
```

Comment:
There is a bug here, the txid you are using in the outPoint here is big endian, NOT little endian which is the protocol level serialization